### PR TITLE
vet fixes

### PR DESCRIPTION
**Pull request by hdonnay**
These are the things that `go vet` complained about. I think the only controversial change (that could be dropped and thought about more) is the `AffectedManifests` commit.

### CODE REVIEW

File: libindex/libindex.go [hunk 1]
```diff
@@ -186,7 +186,7 @@ func (l *Libindex) IndexReport(ctx context.Context, hash claircore.Digest) (*cla
 }
 
 // AffectedManifests retrieves a list of affected manifests when provided a list of vulnerabilities.
-func (l *Libindex) AffectedManifests(ctx context.Context, vulns []claircore.Vulnerability) (claircore.AffectedManifests, error) {
+func (l *Libindex) AffectedManifests(ctx context.Context, vulns []claircore.Vulnerability) (*claircore.AffectedManifests, error) {
```
Comment (ldelossa) [inline libindex/libindex.go:189]: looks good, but was expecting to see some changes in the notifier code, not necessary? Auto de-ref may take care of it without changes.

**Review by ldelossa — APPROVED** [review]

File: libindex/libindex.go (hunk 1 above)
Comment (hdonnay) [reply libindex/libindex.go:189]: What notifier code? This is only used in the libvuln HTTP handler, as far as I can tell.

Comment (ldelossa) [reply libindex/libindex.go:189]: Sorry, I meant I was expecting to see a linked pr: notifier in Clair utilizes this: https://github.com/quay/clair/blob/main/notifier/processor.go

**Review by ldelossa — APPROVED** [review]

**Comment by ldelossa** [comment]
**Note: only backport bug fixes, do not backport api change in libindex/libindex.go**

**Pull request merged**